lib/workers/repository/update/branch/index.ts

```diff
@@ -498,6 +498,9 @@ export async function processBranch(
       }
 
       commitSha = await commitFilesToBranch(config);
+      // Checkout to base branch to ensure that the next branch processing always starts with git being on the baseBranch
+      // baseBranch is not checked out at the start of processBranch() due to pull/16246
+      await checkoutBranch(config.baseBranch);
```

Comment:
This line alone is not enough to fix the problem?

Reply:
Yes, it'll be enough.

Reply:
Then we can revert the others after running on a test repo.

Reply:
Sure, reverted them.